src/lib/media-query/media-observable.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {Injectable} from '@angular/core';
+import {Observable} from 'rxjs/Observable';
+import {map} from 'rxjs/operators/map';
+import {filter} from 'rxjs/operators/filter';
+
+import {MediaChange} from './media-change';
+import {BreakPointRegistry} from './breakpoints/break-point-registry';
+import {MatchMedia} from './match-media';
+import {BreakPoint} from './breakpoints/break-point';
+import {mergeAlias} from '../utils/add-alias';
+
+@Injectable()
+export class MediaObservable {
+
+  /**
+   * Should we announce gt-<xxx> breakpoint activations ?
+   */
+  filterOverlaps = true;
+
+  constructor(private breakpoints: BreakPointRegistry,
+              private matchMedia: MatchMedia) {
+    this._registerBreakPoints();
+  }
+
+  /**
+   * Test if specified query/alias is active.
+   */
+  isActive(alias): boolean {
+    let query = this._toMediaQuery(alias);
+    return this.matchMedia.isActive(query);
+  }
+
+  observe(): Observable<MediaChange> {
+    const activationsOnly = (change: MediaChange) => {
+      return change.matches === true;
+    };
+    const addAliasInformation = (change: MediaChange) => {
+      return mergeAlias(change, this._findByQuery(change.mediaQuery));
+    };
+    const excludeOverlaps = (change: MediaChange) => {
+      let bp = this.breakpoints.findByQuery(change.mediaQuery);
+      return !bp ? true : !(this.filterOverlaps && bp.overlapping);
+    };
+
+    return this.matchMedia.observe()
```

Comment:
the observe() method here is going to get the same comment as I made above... should this be a property? (I'm not saying it *should*, I'm asking... because usually that's the case)

Reply:
In the case of `MatchMedia`, `observe` takes an optional string argument